rpm/src/main/java/org/eclipse/packager/rpm/signature/PgpHeaderSignatureProcessor.java

```diff
@@ -105,6 +110,24 @@ public void feedPayloadData(final ByteBuffer data) {
 
     @Override
     public void finish(final Header<RpmSignatureTag> signature) {
-        signature.putBlob(RpmSignatureTag.RSAHEADER, this.value);
+        switch (this.signature.getKeyAlgorithm()) {
+        // RSA
+        case PublicKeyAlgorithmTags.RSA_GENERAL:
+        case PublicKeyAlgorithmTags.RSA_ENCRYPT:
+        case PublicKeyAlgorithmTags.RSA_SIGN:
+            signature.putBlob(RpmSignatureTag.RSAHEADER, this.value);
+            break;
+
+        // DSA
+        case PublicKeyAlgorithmTags.DSA:
+        case PublicKeyAlgorithmTags.ECDSA:
+        case PublicKeyAlgorithmTags.EDDSA_LEGACY:
```

Comment:
Actually `EDDSA_LEGACY == EDDSA`, so we don't need both. I have verified that the constants we want are `17` and `22` only.